### PR TITLE
Add on(Client)ElementInteriorChange

### LINKED ISSUE #2056: Add onElementInteriorChange
**Is your feature request related to a problem? Please describe.**
We have onElementDimensionChange since r20724, but **onElementInteriorChange is still missing**.

**Describe the solution you'd like**
Add **onElementInteriorChange** and **onClientElementInteriorChange**, which will be triggered when element's interior changed by setElementInterior

### PR DESCRIPTION
**Pull request by gptrk0**
Closes #2056

### CODE REVIEW

File: Server/mods/deathmatch/logic/CElement.h [hunk 1]
```diff
@@ -219,7 +219,7 @@ class CElement
     void RemoveOriginSourceUser(class CPed* pPed) { m_OriginSourceUsers.remove(pPed); }
 
     unsigned char GetInterior() { return m_ucInterior; }
-    void          SetInterior(unsigned char ucInterior) { m_ucInterior = ucInterior; }
+    virtual  void SetInterior(unsigned char ucInterior);
```
Comment (Pirulax) [inline Server/mods/deathmatch/logic/CElement.h]: Why tho? I see no need to make it a virtual method.

**Comment by Pirulax** [comment]
Also, why didnt you make it cancellable? 
It's totally possible to do so.

**Comment by Citizen01** [comment]
> Also, why didnt you make it cancellable?
> It's totally possible to do so.

on(Client)ElementDimensionChanged is not cancelable either and got merged:
https://github.com/multitheftauto/mtasa-blue/commit/7742f34b1fe9b12d3f75d8637e1d73d8fab7d997

Also this event will be triggered even if calling it with the same dimension the entity is already in whereas on(Client)ElementDimensionChanged does not:
```cpp
void CClientEntity::SetDimension(unsigned short usDimension)
{
    if (m_usDimension == usDimension)
        return;

    unsigned int usOldDimension = m_usDimension;
    m_usDimension = usDimension;

    CLuaArguments Arguments;
    Arguments.PushNumber(usOldDimension);
    Arguments.PushNumber(usDimension);
    CallEvent("onClientElementDimensionChange", Arguments, true);
}
```
Both events are meant to be released in 1.5.8 and should act the same way.

If you want it to be cancelable then fine, do it so, but then let's create an issue for the `on(Client)ElementDimensionChanged` event to be cancelable too.

Question: What should happen if the client event is canceled when the interior is changed from the server side ?

**Comment by Patrick2562** [comment]
> Also this event will be triggered even if calling it with the same dimension the entity is already in whereas on(Client)ElementDimensionChanged does not:Yeah, I'm gonna fix it.

**Comment by gptrk0** [comment]
> Also this event will be triggered even if calling it with the same dimension the entity is already in whereas on(Client)ElementDimensionChanged does not:

Yeah, I'm gonna fix it.

> If you want it to be cancelable then fine, do it so, but then let's create an issue for the on(Client)ElementDimensionChanged event to be cancelable too.

We can do it in a different PR, me or someone else.

> Question: What should happen if the client event is canceled when the interior is changed from the server side ?

Dimension changes, doesn't cancel it.

**Comment by Pirulax** [comment]
Actually, let's not add `cancelEvent()`. The problem we run into is that we can't have a consistent behavior (with for example onClientElementDataChange).

**Comment by gptrk0** [comment]
> Actually, let's not add `cancelEvent()`. The problem we run into is that we can't have a consistent behavior (with for example onClientElementDataChange).

I think I reverted everything back to the first commit.

**Review by Pirulax — APPROVED** [review]
LGTM. Test it, if it works I'll merge it.

**Comment by gptrk0** [comment]
@Citizen01 Maybe can you test it if you'll have some time please?

**Comment by PlatinMTA** [comment]
I just tested these changes, and they work. 
[test.zip](https://github.com/multitheftauto/mtasa-blue/files/6150724/test.zip)

I tried:

Using root and an element as a source ✔
Serverside with one object changing interiors every 0,5s ✔
Clientside with one object changing interiors every 0,5s + the trigger from the serverside object ✔
Triggering the event from clientside due to an serverside element ✔

It does seem to work.

**Comment by Pirulax** [comment]
Okay, @Patrick2562 you can go ahead, and merge it IMO.
We'll make the cancellable later on.

**Pull request merged**